src/cache.jl

```diff
@@ -94,7 +94,11 @@ end
 updatecache(absentmodule::Symbol, server) = updatecache([absentmodule], server)
 
 function updatecache(absentmodules::Vector{Symbol}, server)
+    server.cp_lock && return
```

Comment:
Ok, I've not meddled with `Task`s hence my confusion, I'll have a fiddle in the morning by adding a `Channel` to the `LanguageServerInstance`